pptx_blueprint/__init__.py

```diff
@@ -82,6 +84,20 @@ def _find_shapes_in_slide(slide):
 
         return matched_shapes
 
+    def _get_all_shapes(self) -> Iterable[BaseShape]:
+        # Do we need all the shapes? Perhaps we should filter on tags here.
+        all_shapes = [shape for slide in self._presentation.slides for shape in slide.shapes]
+        return all_shapes
+
+    def _copy_tags_to_name(self) -> None:
+        all_shapes = self._get_all_shapes()
+        # This regex matches on tags
+        regex_tag = re.compile(r'\{[\s\w]*\}')
```

Comment:
We should be more strict on the the user-supplied tags.

1. The tag id must match  \w+ (no spaces).
2. We can be permissive and allow whitespace around the tag. - Allowing these can gracefully handle leading or trailing spaces, which might have slipped in unnoticed when editing the template.
3. We shouldn't suuport tags with other text, e.g. `some {tag}` or worse `{tag1} {tab2}`. - At least for now. Handling such things reasonably is much more difficult.

```suggestion
        regex_tag = re.compile(r'^\s*(\{\w+\})\s*$')
```

Reply:
That makes sense. I've applied your suggestions, I hope the PR is usable now :)